packages/@cdktf/hcl2cdk/test/__snapshots__/iteration.test.ts.snap

```diff
@@ -189,37 +203,36 @@ class MyConvertedCode extends cdktf.TerraformStack {
       deletionWindowInDays: 7,
       description: \\"KMS key 1\\",
     });
+    /*In most cases loops should be handled in the programming language context and 
+    not inside of the Terraform context. If you are looping over something external, e.g. a variable or a file input
+    you should consider using a for loop. If you are looping over something only known to Terraform, e.g. a result of a data source
+    you need to keep this like it is.*/
+    const awsS3BucketExamplebucketForEachIterator =
+      cdktf.TerraformIterator.fromList(
+        cdktf.Token.asAny(\`\\\\\${toset(\${buckets.value}.*)}\`)
+      );
     const awsS3BucketExamplebucket = new aws.s3Bucket.S3Bucket(
       this,
       \\"examplebucket\\",
       {
         acl: \\"private\\",
-        bucket: \\"\${each.key}\\",
+        bucket: awsS3BucketExamplebucketForEachIterator.key,
       }
     );
     /*In most cases loops should be handled in the programming language context and 
     not inside of the Terraform context. If you are looping over something external, e.g. a variable or a file input
     you should consider using a for loop. If you are looping over something only known to Terraform, e.g. a result of a data source
     you need to keep this like it is.*/
-    awsS3BucketExamplebucket.addOverride(
-      \\"for_each\\",
-      \`\\\\\${toset(\${buckets.value}.*)}\`
-    );
-    const awsS3BucketObjectExamplebucketObject =
-      new aws.s3BucketObject.S3BucketObject(this, \\"examplebucket_object\\", {
-        bucket: \\"\${each.key}\\",
-        key: \\"someobject\\",
-        kmsKeyId: awsKmsKeyExamplekms.arn,
-        source: \\"index.html\\",
-      });
-    /*In most cases loops should be handled in the programming language context and 
-    not inside of the Terraform context. If you are looping over something external, e.g. a variable or a file input
-    you should consider using a for loop. If you are looping over something only known to Terraform, e.g. a result of a data source
-    you need to keep this like it is.*/
-    awsS3BucketObjectExamplebucketObject.addOverride(
-      \\"for_each\\",
-      \`\\\\\${toset(\${awsS3BucketExamplebucket.fqn}.*)}\`
-    );
+    const awsS3BucketObjectExamplebucketObjectForEachIterator =
+      cdktf.TerraformIterator.fromList(
+        cdktf.Token.asAny(\`\\\\\${toset(\${awsS3BucketExamplebucket.fqn}.*)}\`)
+      );
+    new aws.s3BucketObject.S3BucketObject(this, \\"examplebucket_object\\", {
+      bucket: awsS3BucketObjectExamplebucketObjectForEachIterator.key,
+      key: \\"someobject\\",
+      kmsKeyId: cdktf.Token.asString(awsKmsKeyExamplekms.arn),
+      source: \\"index.html\\",
```

Comment:
(This is true for all examples) For this to work, the iterator has to be passed to the `forEach` property of the resource as well ([related docs with example](https://developer.hashicorp.com/terraform/cdktf/concepts/iterators#when-to-use-iterators)).

Reply:
Yes, did that now.